### PR TITLE
Rework some of the naming code and add tests

### PR DESCRIPTION
**Pull request by danvk**
Follow-up to #14

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -100,20 +93,17 @@ export async function typescriptOfSchema(
     importTs = `import {${symbols}} from "${options.jsonTypesFile}";\n\n`;
   }
 
-  const tableNames = tables.map(t =>
-    normalizeName(getTableTypeName(t, schema, !!options.prefixWithSchemaNames)),
```
Comment (danvk) [inline src/index.ts:104]: Previously the naming logic was duplicated across `typescript.ts` and `index.ts`. Now it's all done in `typescript.ts`.

File: src/typescript.ts [hunk 2]
```diff
@@ -16,24 +16,18 @@ function nameIsReservedKeyword(name: string): boolean {
 }
 
 /**
- * Will determine whether the tableName should be prefixed with the schemaName
+ * Returns a version of the name that can be used as a symbol name, e.g.
+ * 'number' --> 'number_'.
  */
-export function getTableTypeName(
```
Comment (danvk) [inline src/typescript.ts:21]: Looking at the code I realized this wasn't used for type names, just for symbol names. Sorry for the bad suggestion!

File: test/integration/__snapshots__/integration.test.ts.snap [hunk 3]
```diff
@@ -425,11 +444,146 @@ export interface TableTypes {
...
+} as const;
+
+// Table public.table_with_underscores
+export interface PublicTableWithUnderscores {
+  columnWithUnderscores: string;
+}
+export interface PublicTableWithUnderscoresInput {
+  columnWithUnderscores: string;
+}
+const public_table_with_underscores = {
```
Comment (danvk) [inline test/integration/__snapshots__/integration.test.ts.snap:531]: I believe this was the case you mentioned in https://github.com/danvk/pg-to-ts/pull/14/files#r841181807? It seems to be OK here!

**Review by danvk — COMMENTED** [review]
@jamesone FYI

**Comment by jamesone** [comment]
Awesome work!

**Pull request merged**